contrib/fsl_integration/sct_plugin.py

```diff
@@ -208,52 +290,703 @@ def __init__(self, parent):
         self.SetSizerAndFit(self.sizer_h)
 
+    def OnKeyTyped(self, event):
+        txt = event.GetString()
+
     def onButtonGM(self, event):
-        selected_overlay = displayCtx.getSelectedOverlay()
-        filename_path = selected_overlay.dataSource
-        cmd_line = "sct_deepseg_gm -i {} -o seg.nii.gz".format(filename_path)
+
+        overlayORD = displayCtx.overlayOrder
+
+        img1 = overlayORD[0]
+        rawimg = overlayList[img1].dataSource
+        gmimg = self.t1.GetValue()
+
+        print('Raw Image:', rawimg)
+
+        if gmimg == '':
+            print('Output file name not defined, standard name will be used.')
+            base_name = os.path.basename(rawimg)
+            fname, fext = base_name.split(os.extsep, 1)
+            gmname = "{}_gmseg.{}".format(fname, fext)
+            print('Output File Name:', gmname)
+
+            #Standard output file name
+            cmd_line = "sct_deepseg_gm -i {}".format(rawimg)
+            print('Command Line:', cmd_line)
+
+        else:
+            gmname = "{}.nii.gz".format(gmimg)
+            print('Output File Name:', gmname)
+
+            #Personalized output file name
+            cmd_line = "sct_deepseg_gm -i {} -o {}".format(rawimg, gmname)
+            print('Command Line:', cmd_line)
 
         self.call_sct_command(cmd_line)
 
-        outfilename = os.path.join(os.getcwd(), 'seg.nii.gz')
+        outfilename = os.path.join(os.getcwd(), gmname)
         image = Image(outfilename)
         overlayList.append(image)
-
         opts = displayCtx.getOpts(image)
         opts.cmap = 'yellow'
 
+#Computes Cross-Sectional Area for GM, WM and GM+WM (Total)
+class TabPanelCSA(SCTPanel):
+    DESCRIPTION = """This tool computes the spinal cord cross-sectional area. 
+    For more information, please refer to the article below.<br><br>
+    <b>Specific citation</b>:<br>
+    <b>Usage</b>:<br>
+    To launch the script, upload there are two aways to do that. First, upload the raw imaging (T1w, T2w or T2s) into FSLeyes,
+    followed by the spinal cord segmentation. Always keeps this sequence, raw imaging as the first in the Overlay list field,
+    from the bottom to the top, and segmentation imaging as second. Second, just upload the segmentation imaging.
+    If you uploaded more then one image, it is not necessary uploading the images in such order, with the arrows is possible
+    to sort them and only the first imaging will be used. 
+    For more information, please refer to the article below.<br><br>
+    Martin et al.
+    <i>Can Microstructural MRI detect subclinical tissue injury in subjects with asymptomatic
+    cervical spinal cord compression? A prospective cohort study
+    (2018)</i>. BMJ Open. 13;8(4)e01980
+    """
+
+
+    def __init__(self, parent):
+        super(TabPanelCSA, self).__init__(parent=parent,
+                                          id_=wx.ID_ANY)
+        button_gm = wx.Button(self, id=wx.ID_ANY, label="Process Segmentation")
```

Comment:
when running this function on a binary segmentation i get this error (all default fields):
~~~
julien-macbook:~/sct_testing_data $ objc[2500]: Class FIFinderSyncExtensionHost is implemented in both /System/Library/PrivateFrameworks/FinderKit.framework/Versions/A/FinderKit (0x7fff8e5f93d8) and /System/Library/PrivateFrameworks/FileProvider.framework/OverrideBundles/FinderSyncCollaborationFileProviderOverride.bundle/Contents/MacOS/FinderSyncCollaborationFileProviderOverride (0x12b0dbf50). One of the two will be used. Which one is undefined.
The MATPLOTLIBDATA environment variable was deprecated in Matplotlib 3.1 and will be removed in 3.3.
  from matplotlib import cbook
Compute shape analysis:   0%|                           | 0/9 [00:00<?, ?iter/s]
Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_process_segmentation.py", line 286, in <module>
    main(sys.argv[1:])
  File "/Users/julien/code/sct/scripts/sct_process_segmentation.py", line 265, in main
    verbose=verbose)
  File "/Users/julien/code/sct/spinalcordtoolbox/process_seg.py", line 95, in compute_shape
    shape_property = _properties2d(current_patch_scaled, [px, py])
  File "/Users/julien/code/sct/spinalcordtoolbox/process_seg.py", line 154, in _properties2d
    minx, miny, maxx, maxy = region.bbox
TypeError: 'method' object is not iterable
~~~

Reply:
Please, check the order of the segmentation imaging. In the following example is wrong.

Overlay Order: [0, 2, 3, 4, 5, 6, 1]
Segmentation: /Users/julien/sct_testing_data/t2s/t2s.nii.gz

The segmentation imaging must be in second (from the bottom to the top).

Reply:
why do you need the raw image as the first argument? `sct_process_segmentation`'s only mandatory argument is a binary segmentation

Reply:
I was following the our imaging sequence convention. 
However, I think for that case, we could use just the binary segmentation. 
What do you think?